Ex5/GraphNode.java

```diff
@@ -0,0 +1,30 @@
+public class GraphNode <T> {
+	
+	private int inDegree = 0;
```

Comment:
If you changed GraphNode itself to have a list of outgoing edges, you wouldn't need this variable, and wouldn't need the other map either.